### ISSUE TITLE
Bad parameter expansion syntax

### ISSUE THREAD

**Issue by TinCanTech** [koalaman/shellcheck#2297]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

a=1
b=2
c=${${a}:-${b}}
echo $c

```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
No issues detected!

```

#### Here's what I wanted or expected to see:

`c=${${a}:-${b}}` `Bad substitution` - `${a}` is incorrect, use only `a`

**Comment by koalaman** [comment]
ShellCheck has avoided strict validation of parameter expansion because there is no specification and an ever-increasing feature set, so it's tricky to recognize something like `${var@@}` as invalid without breaking on every new bash feature.

However, in this case it starts with another parameter substitution which is never valid.

There are now warnings for anything where the first element of a `${..}` is something completely unexpected, such as another parameter expansion, a command substitution, a quoted variable name, and more, and it catches a number of new issues.

Thanks!
